crates/sui-core/src/consensus_handler.rs

```diff
@@ -646,238 +740,1639 @@ impl<C: CheckpointServiceNotify + Send + Sync> ConsensusHandler<C> {
+        };
+
+        // DONE(commit-handler-rewrite): update transaction status (rejected/finalized) and update metrics
+        let transactions = self.filter_consensus_txns(
+            state.initial_reconfig_state.clone(),
+            &commit_info,
+            &consensus_commit,
+        );
+        // DONE(commit-handler-rewrite): de-duplicate transactions
+        let transactions = self.deduplicate_consensus_txns(&mut state, &commit_info, transactions);
+
+        let mut randomness_manager = self.init_randomness(&mut state, &commit_info);
+
+        // DONE(commit-handler-rewrite): Split transactions into different types for processing.
+        let CommitHandlerInput {
+            user_transactions,
+            capability_notifications,
+            execution_time_observations,
+            checkpoint_signature_messages,
+            randomness_dkg_messages,
+            randomness_dkg_confirmations,
+            end_of_publish_transactions,
+            new_jwks,
+        } = self.build_commit_handler_input(transactions);
+
+        self.process_jwks(&mut state, &commit_info, new_jwks);
+        self.process_capability_notifications(capability_notifications);
+        self.process_execution_time_observations(&mut state, execution_time_observations);
+        self.process_checkpoint_signature_messages(checkpoint_signature_messages);
+
+        let randomness_dkg_updates = self.process_randomness_dkg_messages(
+            randomness_manager.as_deref_mut(),
+            randomness_dkg_messages,
+        );
+
+        let randomness_dkg_confirmation_updates = self.process_randomness_dkg_confirmations(
+            &mut state,
+            randomness_manager.as_deref_mut(),
+            randomness_dkg_confirmations,
+        );
+
+        // DONE(commit-handler-rewrite): [ssm] advance randomness state if needed
+        if randomness_dkg_updates || randomness_dkg_confirmation_updates {
+            if let Some(randomness_manager) = randomness_manager.as_mut() {
+                randomness_manager
+                    .advance_dkg(&mut state.output, commit_info.round)
+                    .await
+                    .expect("epoch ended");
+            }
+        }
+
+        let mut execution_time_estimator = self
+            .epoch_store
+            .execution_time_estimator
+            .try_lock()
+            .expect("should only ever be called from the commit handler thread");
+
+        // DONE(commit-handler-rewrite): load and activate previous round's jwks
+        let authenticator_state_update_transaction =
+            self.create_authenticator_state_update(last_committed_round, &commit_info);
+
+        let (schedulables, randomness_schedulables, assigned_versions) = self.process_transactions(
+            &mut state,
+            execution_time_estimator.as_mut(),
+            &commit_info,
+            authenticator_state_update_transaction,
+            user_transactions,
+        );
+
+        let collected_eop =
```

Comment:
Did we give up cleaning up the rest of this function? They seem to be a lot of clutter, instead of the clean function calls as above.